### ISSUE TITLE
Emission with multiple parameters

### ISSUE THREAD

**Issue by david-sailplan** [1c3t3a/rust-socketio#246]
The Client.emit and Client.emit_with_ack only seem to support a single data parameter

This causes issues for servers who are expecting multiple parameters for the data for these events

rust.rs
```
socket.emit_with_ack("foo", json!("{ack:true}", duration, callback) 
```

example_socketio_server.js
```
socket.on("foo", async (userId, body, callback)=>{})
```
no mater what is placed in data, it will place all data under the "userId" and no way to populate the body parameter.

Is there a workaround that could fix this?

**Comment by 1c3t3a** [comment]
Do you use a special [parser](https://socket.io/docs/v4/custom-parser/) for this on the server side?

**Comment by david-sailplan** [comment]
No. We don't use any parser beyond the default behavior of the nodejs package server-side.

`const io = socketio(server, {
    path: '/socket',
    transports: ['websocket']
  })`

**Comment by SSebo** [comment]
@david-sailplan https://github.com/1c3t3a/rust-socketio/blob/4ac27c7158a89375c503164857d86b1a1167116c/socketio/src/socket.rs#L105-L109 currently is limit to only two params, there is no workaround except add new function I think.

**Comment by syshriki9** [comment]
@SSebo is there currently a better way to do this using "binary data/attachments"? 

I know your todo list is probably long, but a proper interface for encoding multiple message parameters would be a great feature :)

**Comment by SSebo** [comment]
cc @1c3t3a @nshaaban-cPacket 

> @SSebo is there currently a better way to do this using "binary data/attachments"?
> 
> I know your todo list is probably long, but a proper interface for encoding multiple message parameters would be a great feature :)

cc @1c3t3a @nshaaban-cPacket

**Comment by SSebo** [comment]
#249 I made a PR @david-sailplan You can have a try.

**Comment by syshriki9** [comment]
Wow that was fast!, Thanks @SSebo.

As David mentioned, it seems to work.

**Comment by syshriki9** [comment]
@SSebo @1c3t3a do you guys think this feature will make it into the master?

**Comment by shenjackyuanjie** [comment]
also. that……
hmmm
well, now the `.on` can not accept mult payload, so ……
anther old issue

**Comment by shenjackyuanjie** [comment]
This issue is mostly done, likely able to close
